tests/flow/test_algebraic_expression_crash.py

```
@@ -0,0 +1,32 @@
+    def test_optional_match_with_index(self):
+        # This query caused a crash in AlgebraicExpression_Dest
+        query = """MATCH (n2 :L5{k21:'txLpbrhL'})-[r1]->(n3)  OPTIONAL MATCH (n2 :L3)-[r6]->(n3) RETURN DISTINCT *"""
+        
+        # The query should complete without crashing
+        # We don't care about the result (which should be empty since no data exists)
+        # We just care that it doesn't crash
+        try:
+            result = self.graph.query(query)
+            # If we got here, the crash was avoided
+            self.env.assertTrue(True)
+        except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)